tests/schema_update_test.go

```diff
@@ -543,3 +545,322 @@ func TestVCS(t *testing.T) {
+					Project: gitlab.WebhookProject{
+						ID: 121,
+					},
+					CommitList: []gitlab.WebhookCommit{
+						{
+							Timestamp: "2021-01-13T13:14:00Z",
+							AddedList: []string{gitFile},
+						},
+					},
+				}
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			port := getTestPort(t.Name())
+			mysqlPort := port + 3
+			a := require.New(t)
+			ctx := context.Background()
+			ctl := &controller{}
+			err := ctl.StartServer(ctx, t.TempDir(), test.vcsProviderCreator, port)
+			a.NoError(err)
+			defer func() {
+				_ = ctl.Close(ctx)
+			}()
+
+			err = ctl.Login()
+			a.NoError(err)
+			err = ctl.setLicense()
+			a.NoError(err)
+
+			// Create a VCS.
+			vcs, err := ctl.createVCS(
+				api.VCSCreate{
+					Name:          t.Name(),
+					Type:          test.vcsType,
+					InstanceURL:   ctl.vcsURL,
+					APIURL:        ctl.vcsProvider.APIURL(ctl.vcsURL),
+					ApplicationID: "testApplicationID",
+					Secret:        "testApplicationSecret",
+				},
+			)
+			a.NoError(err)
+
+			// Create a project.
+			project, err := ctl.createProject(
+				api.ProjectCreate{
+					Name: "Test VCS Project",
+					Key:  "TVP",
+				},
+			)
+			a.NoError(err)
+
+			// Create a repository.
+			ctl.vcsProvider.CreateRepository(externalID)
+			_, err = ctl.createRepository(
+				api.RepositoryCreate{
+					VCSID:              vcs.ID,
+					ProjectID:          project.ID,
+					Name:               "Test Repository",
+					FullPath:           repoFullPath,
+					WebURL:             fmt.Sprintf("%s/%s", ctl.vcsURL, repoFullPath),
+					BranchFilter:       branchFilter,
+					BaseDirectory:      baseDirectory,
+					FilePathTemplate:   test.filePathTemplate,
+					SchemaPathTemplate: "{{ENV_NAME}}/.{{DB_NAME}}__LATEST.sql",
+					ExternalID:         externalID,
+					AccessToken:        "accessToken1",
+					RefreshToken:       "refreshToken1",
+				},
+			)
+			a.NoError(err)
+
+			// environments, err := ctl.getEnvironments()
+			// a.NoError(err)
+			// prodEnvironment, err := findEnvironment(environments, "Prod")
+			// a.NoError(err)
+			environment, err := ctl.createEnvrionment(api.EnvironmentCreate{
+				Name: test.envName,
+			})
+			a.NoError(err)
+			// Provision an instance.
+			instanceName := "testInstance"
+			_, stopInstance := resourcemysql.SetupTestInstance(t, mysqlPort)
+			defer stopInstance()
+			connCfg := getMySQLConnectionConfig(strconv.Itoa(mysqlPort), "")
+			instance, err := ctl.addInstance(api.InstanceCreate{
+				EnvironmentID: environment.ID,
+				Name:          instanceName,
+				Engine:        db.MySQL,
+				Host:          connCfg.Host,
+				Port:          connCfg.Port,
+				Username:      connCfg.Username,
+			})
+			a.NoError(err)
+
+			// Create an issue that creates a database.
+			err = ctl.createDatabase(project, instance, dbName, "", nil /* labelMap */)
+			a.NoError(err)
+
+			a.Equal(len(test.commitFileNames), len(test.expect))
```

Comment:
The expectation comes first.

Reply:
We need to make sure their length are equal.

Reply:
I feel like you two are speaking on different things 😂

Reply:
> I feel like you two are speaking on different things 😂

oh...


> The expectation comes first.

The `expect` is not the expectation argument of `a.Equal()` function.

Reply:
func Equal(t [TestingT](https://pkg.go.dev/github.com/stretchr/testify@v1.8.0/assert?utm_source=gopls#TestingT), expected, actual interface{}, msgAndArgs ...interface{}) [bool](https://pkg.go.dev/builtin#bool)

Let's change `a.Equal(len(test.commitFileNames), len(test.expect))` to `a.Equal(len(test.expect), len(test.commitFileNames))` if `len(test.expect)` is what we want.